### PR TITLE
Add achainable Account&Balance&Dotsama&DeFi Tag apis

### LINKED ISSUE #1718: Create new assertions based on current new Achainable API (tag1: Account)
### Context
Our data provider TDF's product [Achainable](https://achainable-stage.vercel.app/?achainableLabels=true&userLabels=true&searchText=&chain=All+chains&tag=All+tags&view=list) has developed new API to query data. There are quite many labels provided by Achainable. They are grouped by tags: Account, Balance,  Dotsama, Bridge, etc.

### Task
Covert existed labels into assertions. Start from tag: Account.

### Acceptance criteria
What's the definition of done, if applicable?

---
:heavy_check_mark: Please set appropriate **labels** and **assignees** if applicable.

### PR DESCRIPTION
**Pull request by outofxxx**
cover part of:  #1718, #1719, #1720, #1826 

- [x] Tag: Account
- [x] Tag: Balance
- [x] Tag: Dotsama
- [x] Tag: DeFi

BillyWooo: There are some overlap between existed assertions, new Achainable APIs and new assertions (which should be added). So let's close this PR first, then create another new PR to add new assertions.

### CODE REVIEW

File: tee-worker/litentry/core/data-providers/src/graphql.rs [hunk 1]
```diff
@@ -264,6 +265,224 @@ pub struct TotalTxsStruct {
...
+	fn is_kusama_validator(&mut self, address: &str) -> Result<bool, Error>;
+}
+
+impl AchainableTagAccount for GraphQLClient {
+	fn fresh_account(&mut self, address: &str) -> Result<bool, Error> {
+		let params = ReqParams::new("/v1/run/label/1de85e1d215868788dfc91a9f04d7afd");
+		let body = ReqBody { params: ParamsAccount { address: address.to_string() } };
+		let response =
+			self.client.post_capture::<ReqParams, ReqBody, serde_json::Value>(params, &body);
+		debug!("ReqBody response: {:?}", response);
```
Comment (kziemianek) [inline tee-worker/litentry/core/data-providers/src/graphql.rs]: These functions differs only in request's params and body, have you considered extracting the common part ?

File: tee-worker/litentry/core/data-providers/src/graphql.rs [hunk 2]
```diff
@@ -264,6 +265,224 @@ pub struct TotalTxsStruct {
...
+/// base_url: https://label-production.graph.tdf-labs.io
+/// test key: 26353d4c-b01c-4466-98a5-80d3fc53a9d8
+pub trait AchainableTagAccount {
+	fn fresh_account(&mut self, address: &str) -> Result<bool, Error>;
+	fn og_account(&mut self, address: &str) -> Result<bool, Error>;
+	fn class_of_2020(&mut self, address: &str) -> Result<bool, Error>;
+	fn class_of_2021(&mut self, address: &str) -> Result<bool, Error>;
+	fn class_of_2022(&mut self, address: &str) -> Result<bool, Error>;
+	fn found_on_bsc(&mut self, address: &str) -> Result<bool, Error>;
+	fn is_polkadot_alidator(&mut self, address: &str) -> Result<bool, Error>;
```
Comment (kziemianek) [inline tee-worker/litentry/core/data-providers/src/graphql.rs]: `is_polkadot_validator` 🙃

File: tee-worker/litentry/core/data-providers/src/graphql.rs (hunk 1 above)
Comment (outofxxx) [reply tee-worker/litentry/core/data-providers/src/graphql.rs]: Yes, there are only slight differences between them. Subsequent pr will include all Balance-related interfaces, and the corresponding refactoring is planned on that pr.

**Comment by kziemianek** [comment]
Are you sure this PR is closing the corresponding issue?
I don't see any new assertions added.

File: tee-worker/litentry/core/data-providers/src/graphql.rs (hunk 1 above)
Comment (BillyWooo) [reply tee-worker/litentry/core/data-providers/src/graphql.rs]: I would still suggest to do a refactor now. Left to next PR will create more complexity. As I see, only extract one "params", then the common part can be used.

**Review by BillyWooo — CHANGES_REQUESTED** [review]
I think all "GraphQL" name needs to be changed. "Achainable" or something you think fits. Overall we might have different data provider in the future. GraphQL will create confusion. 
Besides, do we have any test for new code?

**Comment by Kailai-Wang** [comment]
Is it related to #1654 too?

**Comment by outofxxx** [comment]
> Is it related to #1654 too?

#1654 is not included, only the new API under Account Tag is included;

File: tee-worker/service/src/running-mode-config.json [hunk 3]
```diff
@@ -43,8 +43,8 @@
         "discord_official_url": "https://discordapp.com",
         "discord_litentry_url": "http://54.255.182.249:9527",
         "discord_auth_token": "ABCDEFGHIJKLMNOPQRSTUVWXYZ",
-        "graphql_url": "https://graph.tdf-labs.io/",
-        "graphql_auth_key": "ac2115ec-e327-4862-84c5-f25b6b7d4533",
+        "achainable_url": "https://graph.tdf-labs.io/",
+        "achainable_auth_key": "ac2115ec-e327-4862-84c5-f25b6b7d4533",
```
Comment (kziemianek) [inline tee-worker/service/src/running-mode-config.json]: Is this the real value used in staging env or just placeholder?
Maybe we will need to update deployment env as well.

Comment (BillyWooo) [reply tee-worker/service/src/running-mode-config.json]: They are just placeholders. Real value is only stored on our server.

Comment (kziemianek) [reply tee-worker/service/src/running-mode-config.json]: @BillyWooo do we need to make configuration changes on server before deploying this change ?

File: tee-worker/litentry/core/data-providers/src/achainable.rs [hunk 4]
```diff
@@ -0,0 +1,1567 @@
...
+	fn verified_credentials_is_hodler(&mut self, query: Query) -> Result<IsHodlerOut, Error> {
+		let path = query.path();
+		let query_value = query.to_achainable();
+		debug!("verified_credentials_is_hodler query: {}", query_value);
+
+		let query = vec![("query", query_value.as_str())];
+		match self.client.get_with::<String, QLResponse>(path, query.as_slice()) {
+			Ok(response) =>
+				if let Some(value) = response.data.get("data") {
+					debug!("	[Graphql] value: {:?}", value);
```
Comment (Kailai-Wang) [inline tee-worker/litentry/core/data-providers/src/achainable.rs]: Seems we missed this

File: tee-worker/litentry/core/data-providers/src/achainable.rs [hunk 5]
```diff
@@ -0,0 +1,1567 @@
...
+				)
+			})
+			.collect::<Vec<String>>();
+		format!("query {{{}}}", q.join(","))
+	}
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct QLResponse {
```
Comment (Kailai-Wang) [inline tee-worker/litentry/core/data-providers/src/achainable.rs]: Maybe simply `Response`? It's going to be changed anyway

**Review by Kailai-Wang — APPROVED** [review]
I didn't review every single line, but if CI passes it's mostly fine.

Although:
- I only see the achainable client implements several traits, but I don't see the actual assertion building. What should I do if I want a `polkadot_dolphin` VC?
- the `achainable.rs` has a significant amount of duplicate code, we should be able to simplify it (e.g. put the mapping of label name and achainable URL into a JSON), or use macros

File: tee-worker/litentry/core/data-providers/src/achainable.rs (hunk 5 above)
Comment (BillyWooo) [reply tee-worker/litentry/core/data-providers/src/achainable.rs]: renamed to `AchainableResponse`

File: tee-worker/litentry/core/data-providers/src/achainable.rs (hunk 4 above)
Comment (BillyWooo) [reply tee-worker/litentry/core/data-providers/src/achainable.rs]: changed to `Achainable`.

**Comment by BillyWooo** [comment]
> * the `achainable.rs` has a significant amount of duplicate code, we should be able to simplify it (e.g. put the mapping of label name and achainable URL into a JSON), or use macros

made a simple version of refactor.

**Comment by kziemianek** [comment]
@BillyWooo I don't see any new assertions created in this PR, we are only replacing old API with new. Am I missing something ? Issue #1718 is about creating new assertions or just migrating to new API ?

**Comment by BillyWooo** [comment]
> @BillyWooo I don't see any new assertions created in this PR, we are only replacing old API with new. Am I missing something ? Issue #1718 is about creating new assertions or just migrating to new API ?

I think you are right. But not to all of them. There are some overlap with existed assertions. But yes, new assertions should be added.

**Comment by kziemianek** [comment]
> > @BillyWooo I don't see any new assertions created in this PR, we are only replacing old API with new. Am I missing something ? Issue #1718 is about creating new assertions or just migrating to new API ?
> 
> I think you are right. But not to all of them. There are some overlap with existed assertions. But yes, new assertions should be added.

Just wanted to point out that this pull request is not fixing corresponding issues.

**Review by BillyWooo — APPROVED** [review]

**Pull request merged**